agent/skyhook-agent/tests/test_controller.py

```
@@ -1104,6 +1104,26 @@ def test_interrupt_applies_all_commands(self, run_mock, datetime_mock):
                 mock.call(root_dir, ["systemctl", "restart", "containerd"], controller.get_log_file("interrupts/service_restart_1", copy_dir, config_data, root_dir), copy_dir=copy_dir, write_cmds=True, no_chmod=True)
             ])
 
+    @mock.patch("skyhook_agent.controller.do_interrupt")
+    def test_interrupt_bootstraps_copy_dir_before_running(self, do_interrupt_mock):
+        def _assert_copy_dir_exists(interrupt_data, root_mount, copy_dir):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_assert_copy_dir_exists`

Add return type annotation: `bool`

(ANN202)